File: functions-v4-python-modelv2/src/QueueStorage.py
import logging 

import azure.functions as func

import uuid

bp = func.Blueprint() 

@bp.route(route="QueueStorageOutput1", auth_level=func.AuthLevel.FUNCTION)
@bp.queue_output(arg_name="msg", 
                  queue_name="functions-testkit-queue", 
                  connection="QueueStorageConnectionString")
def QueueStorageOutput1(req: func.HttpRequest, msg: func.Out[str]) -> func.HttpResponse:
    logging.info('Python HTTP trigger function processed a request.')

    name = req.params.get('name')
    if not name:
        try:
            req_body = req.get_json()
        except ValueError:
            pass
        else:
            name = req_body.get('name')

    # queue storage output
    line = f"{uuid.uuid4()}"
    logging.info(f"[debug] QueueStorage output: {line}")
    msg.set( line )

    ### it is not work. queue output only support single message.
    # for i in range(3):
    #     line = f"{i}:{uuid.uuid4()}"
    #     logging.info(f"[debug] QueueStorage output: {line}")
    #     msg.set(line)


    if name:
        return func.HttpResponse(f"Hello, {name}. This HTTP triggered function executed successfully.")
    else:
        return func.HttpResponse(
             "This HTTP triggered function executed successfully. Pass a name in the query string or in the request body for a personalized response.",
             status_code=200
        )


@bp.function_name(name="QueueStorageTrigger1")
@bp.queue_trigger(arg_name="msg", queue_name="functions-testkit-queue",
                   connection="QueueStorageConnectionString")  # Queue trigger
def QueueStorageTrigger1(msg: func.QueueMessage) -> None:
    logging.info('Python queue trigger function processed a queue item: %s',
                 msg.get_body().decode('utf-8'))

File: functions-v4-typescript-modelv4/src/functions/QueueStorage.ts
import { app, InvocationContext } from "@azure/functions";
import { HttpRequest, HttpResponseInit, output } from "@azure/functions";


const queueOutput = output.storageQueue({
    queueName: 'outqueue',
    connection: 'MyStorageConnectionAppSetting',
});
export async function QueueStorageOutput1(request: HttpRequest, context: InvocationContext): Promise<HttpResponseInit> {
    context.log(`Http function processed request for url "${request.url}"`);

    context.extraOutputs.set(queueOutput, "Hello");
    const name = request.query.get('name') || await request.text() || 'world';

    return { body: `Hello, ${name}!` };
};

app.http('QueueStorageOutput1', {
    methods: ['GET', 'POST'],
    authLevel: 'anonymous',
    handler: QueueStorageOutput1
});



export async function QueueStorageTrigger1(queueItem: unknown, context: InvocationContext): Promise<void> {
    context.log('Storage queue function processed work item:', queueItem);
}

app.storageQueue('QueueStorageTrigger1', {
    queueName: 'js-queue-items',
    connection: '',
    handler: QueueStorageTrigger1
});


